src/core/DataStream.jl

```diff
@@ -176,6 +192,32 @@ function evolve_payload_range!(
         round((stream.initial_range.b + growth) * stream.size_ratio)
 end
 
+struct StepAssociatedFunction
+    step::Integer
+    fn::EvolutionFunction
+end
+
+StepAssociatedFunction(pcfg::Config) =
+    StepAssociatedFunction(pcfg[:step], get_evolution_function(pcfg))
+
+struct PiecewiseEvFn <: EvolutionFunction
+    fns::Vector{StepAssociatedFunction}
+    PiecewiseEvFn(params::Vector{T}) where {T<:Config} =
```

Comment:
I believe Julia moved from constructors+new to proper functions with the struct name, just a comment. We can refactor later.